src/main/java/org/jabref/gui/frame/JabRefFrameViewModel.java

```diff
@@ -180,6 +202,29 @@ public void handleUiCommands(List<UiCommand> uiCommands) {
                   });
     }
 
+    private Optional<Path> firstBibFile() {
+        Path currentDir = Path.of("").toAbsolutePath();
+
+        while (currentDir != null) {
+            try {
+                Optional<Path> bibFile = Files.list(currentDir)
```

Comment:
this looks horrible. Did you use chatgpt?  Better use Files.find  that does all this under the hood already
https://mkyong.com/java/java-files-find-examples/

Reply:
Also climbing up!!!

I search x/y/z. If there not found, go one dir up.

I can also do recursion...

I do NOT want list all files form root (e.g.., / or c:\) and then filter. This takes too long!!

Reply:
> I search x/y/z. If there not found, go one dir up.

This does not make sense. 
I assume a scenario with JabRef portable. You have the bib file next to JabRef.exe and it will load it. This is fine.
But not randomly searching folders up and up or down

And additionally, this features makes so sense under Mac as you have the Appplications folder and all .app apps.
You would never place a file there.

Reply:
My use case:

![image](https://github.com/user-attachments/assets/552a9d32-5692-4d05-bee8-dac3c4878fc0)

![image](https://github.com/user-attachments/assets/41b9f3ee-3bec-4d85-a12c-83ce5d149f27)

Reply:
![image](https://github.com/user-attachments/assets/cfc54353-bed3-4c5c-af58-ccf73ee86051)